seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/utils/HiveMetaStoreProxy.java

```diff
@@ -158,6 +159,36 @@ public Table getTable(@NonNull String dbName, @NonNull String tableName) {
         }
     }
 
+    public void createDatabaseIfNotExists(String db) throws TException {
```

Comment:
How about let `HiveMetaStoreProxy` implements Catalog interface? So that we do need overwrite most method of `HiveSaveModeHandler`.